app/feeswapper/management/commands/tests/test_feeSwapper.py

```diff
@@ -0,0 +1,34 @@
+import json
+from unittest import mock
+
+from django.core.management import call_command
+
+from feeswapper.management.commands.feeSwapper import Command
+from feeswapper.models import CurrencyConversion
+from test_plus.test import TestCase
+
+
+class RequestsMockResponse:
+    def __init__(self, status_code, json_data=None):
+        self.json_data = json_data
+        self.status_code = status_code
+        self.ok = True
+
+    def json(self):
+        return self.json_data
+
+class feeSwapperTest(TestCase):
+    """Define tests for feeSwapper management command
+    
+    """
+    resp = RequestsMockResponse(200,json_data={"status":"1","result":[{"symbol":"OMG","name":"OMGToken","decimals":"18","contractAddress":"0x879884c3c46a24f56089f3bbbe4d5e38db5788c0","balance":"10000000"},{"symbol":"WEENUS","name":"Weenus 💪","decimals":"18","contractAddress":"0xaff4481d10270f50f203e0763e2597776068cbc5","balance":"1000000000000000000000"},{"symbol":"BAT","name":"Basic Attention Token","decimals":"18","contractAddress":"0xda5b056cfb861282b4b59d29c9b395bcc238d29b","balance":"0"},{"symbol":"MKR","name":"Maker","decimals":"18","contractAddress":"0xf9ba5210f91d0474bd1e1dcdaec4c58e359aad85","balance":"100000000000000000"}],"message":"OK"})
+    
+    @mock.patch('feeswapper.management.commands.feeSwapper.requests.get',return_value=resp)
+    @mock.patch('feeswapper.management.commands.feeSwapper.Command.factoryContract', return_value="0x0000000000000000000000000000000000000001")
+    @mock.patch('feeswapper.management.commands.feeSwapper.Command.web3', return_value='1')
+    def test_getTokenList(self, response, web3_response, toWei_response):
```

Comment:
Is this the only test? I would like to see more tests given the other parts of this PR are much more complicated than `getTokenList`

Reply:
I spent a while trying to come up with a useful way to write a unit test for sell_token since alot of it is web3 calls to the uniswap contracts.  I was having to mock all those responses so it didn't feel like much of a useful test.  My initial look at the available docs on eth-tester library didn't seem to provide much help here either.  I'm not super familiar with web3 as this was my first real coding work that involved it so not sure if there are obvious ways to unit test that library that I'm missing.
Are you also looking for tests that validate the negative scenarios/database checks?

Reply:
I've added in additional tests for the 2 major scenarios in the sell_token function.
The first one validates that it doesn't try to exchange a token where there is an existing record in the DB for a failed transaction.  The second tests the math in the code that conducts the exchange on Uniswap.  I had to mock all the web3 calls related to the Uniswap contract interaction but this will validate that the local code does the math right and puts the right numbers in the DB record.  

I gotta say, web3 calls are wicked to mock.